corehq/apps/userreports/templates/userreports/expression_debugger.html

```diff
@@ -56,20 +56,31 @@ <h1>{% trans "UCR Expression Debugger" %}</h1>
         </div>
       </div>
     {% endif %}
-    <div class="form-group">
+    <div class="form-group" data-bind="css: {'has-error': hasParseError}">
       <label for="" class="col-sm-2 control-label">{% trans "Expression JSON" %}</label>
       <div class="col-sm-10">
         <textarea id="expression" class="form-control jsonwidget" ></textarea>
+        <div class="help-block" data-bind="visible: hasParseError">
+          {% blocktrans %}
+            Your expression has parse errors!
+            For more details, try pasting into a <a href="http://jsonlint.com/" target="_blank">JSON Validator</a>.
+          {% endblocktrans %}
+        </div>
       </div>
     </div>
     <div class="form-group">
       <div class="col-sm-offset-2 col-sm-10">
-        <button type="submit" class="btn btn-primary">{% trans "Evaluate!" %}</button>
+        <button type="submit" class="btn btn-primary">
+          {% trans "Evaluate!" %}
+        </button>
+      </div>
+    </div>
+    <div class="col-sm-offset-2 col-sm-10">
+      <div class="alert alert-danger" data-bind="visible: error, html: error"></div>
+      <div class="alert alert-success" data-bind="visible: result">
+        {% trans "Success!" %}
+        <pre data-bind="text: result"></pre>
```

Comment:
it was not clear to me in the example images on PR as well, what does the result signify? 
the example had the number 2. May be some text here can help like 2 cases or 2 forms or 2 something.

Reply:
The expressions can signify virtually anything, so I don't think there's an accurate way to show more context. In this case, the `2` signifies is a calculation done on a case property.